editor/test/editor/prefs_test.clj

```diff
@@ -1,37 +1,234 @@
-;; Copyright 2020-2024 The Defold Foundation
-;; Copyright 2014-2020 King
-;; Copyright 2009-2014 Ragnar Svensson, Christian Murray
-;; Licensed under the Defold License version 1.0 (the "License"); you may not use
-;; this file except in compliance with the License.
-;; 
-;; You may obtain a copy of the License, together with FAQs at
-;; https://www.defold.com/license
-;; 
-;; Unless required by applicable law or agreed to in writing, software distributed
-;; under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
-;; CONDITIONS OF ANY KIND, either express or implied. See the License for the
-;; specific language governing permissions and limitations under the License.
-
 (ns editor.prefs-test
-  (:require [clojure.test :refer :all]
-            [editor.prefs :as prefs])
-  (:import [javafx.scene.paint Color]))
+  (:require [clojure.edn :as edn]
+            [clojure.test :refer :all]
+            [editor.fs :as fs]
+            [editor.prefs :as prefs]))
+
+(prefs/register-schema!
+  ::types
+  {:type :object
+   :properties {:types {:type :object
+                        :properties {:any {:type :any}
+                                     :boolean {:type :boolean :default true}
+                                     :string {:type :string}
+                                     :keyword {:type :keyword :default :none}
+                                     :integer {:type :integer :default -1}
+                                     :number {:type :number :default 0.5}
+                                     :array {:type :array :item {:type :string}}
+                                     :set {:type :set :item {:type :string}}
+                                     :enum {:type :enum :values [:foo :bar]}
+                                     :tuple {:type :tuple :items [{:type :string} {:type :keyword :default :code-view}]}}}}})
```

Comment:
It might be nice to have some sort of macro or `AutoClosable` thing that deregisters the schemas after their respective tests conclude. There's a bit of potential for cross-contamination between tests here. 🤔